Delphi.Mocks.Helpers.pas

```diff
@@ -102,6 +104,43 @@ function CompareValue(const Left, Right: TValue): Integer;
   end;
 end;
 
+function SameVariant(const Left, Right: Variant): Boolean;
+var
+  i: integer;
+begin
+  // here, we are going to check to the best of our ability that the variants are the same.
+  // simply a = b is not enough in some cases with variants
+
+  // a varArray of variants needs to be correctly checked, as implicit comparison does not work
+  if ( VarType(Left) = VarType(Right) ) AND
+     ( VarType(Left) = varArray + varVariant ) then
```

Comment:
This check will only work for array of Variant but fail for any other array inside a variant (you can test that if you create a varArray with something else than varVariant - like varInteger). Imo it should be (VarType(Left) and varArray = varArray)